sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubsTransportType.cs

```diff
@@ -14,6 +14,9 @@ public enum EventHubsTransportType
         AmqpTcp,
 
         /// <summary>The connection uses the AMQP protocol over web sockets.</summary>
-        AmqpWebSockets
+        AmqpWebSockets,
+
+        /// <summary>The connection uses the AMQP protocol over TCP without TLS</summary>
```

Comment:
In order to signal the SDK about using Event Hubs local test environment just by connection string and not introducing any additional transport type and client-side changes, I can think of following:

**- Option 1: New Connection Parameter** 

Considering following connection strings:
- Environment without TLS (Current State and Default Environment Behavior):
var connectionString = "Endpoint=sb://contoso.servicebus.local.net/;SharedAccessKeyName=SomeKeyName;SharedAccessKey=SomeKeyValue;LocalEnvironmentConnectionMode=NonTls";

- Environment with TLS Support (In Future) -> Essentially same as current behavior:
var connectionString = "Endpoint=sb://contoso.servicebus.local.net/;SharedAccessKeyName=SomeKeyName;SharedAccessKey=SomeKeyValue;LocalEnvironmentConnectionMode=Tls";

There won't be any enforcement over the host endpoint for any pattern. Using"LocalEnvironmentConnectionMode=NonTls" user can indicate they wants to interact with the Test Environment. 

But this creates a lever in SDK where user can send non-TLS requests to the cloud service by using with any prod connection string.


**- Option 2: Connection String Suffix Regex Match** 

Considering the following Endpoint param in a connection string:
sb://contoso.servicebus.local-v1.net 

If the endpoint matches the pattern where URI ends with ".local-v1.net" we can assume it as an Environment without TLS.
In future when test environment supports TLS the endpoint it will generate will change to "local-v2" thereby not matching the pattern and going with the default behavior with TLS. 

@jsquire  please do share your suggestions.